homeassistant/util/async_.py

```diff
@@ -130,15 +130,15 @@ def check_loop() -> None:
         extra = ""
 
     _LOGGER.warning(
-        "Detected I/O inside the event loop. This is causing stability issues. Please report issue%s for %s doing I/O at %s, line %s: %s",
+        "Detected blocking call inside the event loop. This is causing stability issues. Please report issue%s for %s doing blocking calls at %s, line %s: %s",
         extra,
         integration,
         found_frame.filename[index:],
         found_frame.lineno,
         found_frame.line.strip(),
     )
     raise RuntimeError(
-        f"I/O must be done in the executor; Use `await hass.async_add_executor_job()` "
+        f"Blocking calls must be done in the executor; Use `await hass.async_add_executor_job()` "
```

Comment:
To be strict we don't want calls to sleep in our thread pool either, since it may exhaust the pool, but it's definitely better to do it there than in the event loop. It's ok to sleep in a separate thread, ie neither the event loop or part of our thread pool. Not sure how much we want to explain in this exception message.

Reply:
Would you have a suggestion for a better worded log message?

Reply:
Something like this. Note I dropped not needed f-string.
```suggestion
        "Blocking calls must be done in the executor or a separate thread; "
        "Use `await hass.async_add_executor_job()` "
```